bindist/linux/debian/build_deps.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python3
+import os
+import shutil
+import subprocess
+import sys
+import tempfile
+import glob
+import json
+import getpass
+
+import prepare_deps
+
+HACKDIR = "/homedoesnotexistatbuildtime/"
+
+OLD_STYLE_CABAL = [
+    "time-compat",
+    "type-equality",
+    "binary-orphans"
+]
+
+HERE = os.path.abspath(os.path.dirname(__file__))
+ROOT = os.path.abspath(os.path.join(HERE, "../../.."))
+CACHE_DIR = os.path.expanduser(os.path.join(HERE, "deb_cache"))
+
+
+def install_debs(target_dir, package):
+    debs = list(glob.glob(os.path.join(target_dir, "*.deb")))
+    subprocess.run(["sudo", "dpkg", "-i"] + debs, check=True)
+    return debs
+
+
+def build_deb(tmpdir, package, src_dir, force=False):
+    print(json.dumps(package, indent=2))
+    target_dir = os.path.join(CACHE_DIR, os.path.basename(src_dir))
+
+    if force:
+        shutil.rmtree(target_dir, ignore_errors=True)
+    elif os.path.exists(target_dir):
+        return False, install_debs(target_dir, package)
+
+    tmp_target = os.path.join(tmpdir, os.path.basename(src_dir))
+    shutil.copytree(src_dir, tmp_target)
+    os.chdir(tmp_target)
+
+    cabal_debian_cmd = [
+        "cabal-debian",
+        "--native",
+        "--disable-tests",
+        "--dep-map", "pthread:libpthread-stubs0-dev"
+    ]
+
+    # clash-prelude work-around
+    if package["pkg-name"] in {"clash-prelude", "clash-lib", "clash-ghc"}:
+        # TODO: Ask Leon
```

Comment:
@leonschoorl I'm currently disabling haddock for `clash*` due to it failing on 8.6. Can I add something to the `cabal.project` that forces a newer haddock version, or is it tightly coupled with GHC?

Reply:
Looking at http://hackage.haskell.org/package/haddock-api-2.23.0, it seems pretty tightly bound to the GHC api version. Sadly the changelog doesn't say which version contains the fix for loading plugins.

Reply:
Alright, that's too bad. I guess we'll have to wait for the docker image of `Ubuntu 20.04 ‘Focal Fossa’` then. They seem to update GHC pretty consistently with every Ubuntu version.

Reply:
And yes, the haddock version is tightly coupled with GHC version.

Reply:
Too bad. Thanks for looking into it anyway!